### PR TITLE
add ability to inject environment variables globally into launched processes

### PR DESCRIPTION
**Pull request by jrhizor**
Specifically this will enable us to inject values like the `SENTRY_DSN` when it is set in the `.env` files or Helm. It will also enable us to do things like set `LOG_LEVEL` (if it's respected at least) for launched containers in Docker and Kubernetes.

This partially addresses https://github.com/airbytehq/airbyte/issues/9104 (we can do a separate PR will actually introduce the sentry tracking -- I'd prefer for someone on connectors to do this to get a sense of where these overrides have to be added on OSS and cloud).

### CODE REVIEW

**Review by lmossman — APPROVED** [review]
Just had one small question but overall looks good to me!

File: airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/OrchestratorConstants.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
       EnvConfigs.JOB_MAIN_CONTAINER_CPU_LIMIT,
       EnvConfigs.JOB_MAIN_CONTAINER_MEMORY_REQUEST,
       EnvConfigs.JOB_MAIN_CONTAINER_MEMORY_LIMIT,
+      // EnvConfigs.JOB_ENV_MAP,
```
[the remark this answers is not in the document]
Comment (jrhizor) [reply airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/OrchestratorConstants.java:31]: Gah I was testing it at the end turning things off and on again and forgot to fix it.

**Comment by michel-tricot** [comment]
Have you considered this pattern instead of serializing json in env variables?
```
JOB_DEFAULT_ENV_[KEY1]=abc
JOB_DEFAULT_ENV_[KEY2]=def
...
```

**Review by sherifnada — APPROVED** [review]

**Review by cgardens — APPROVED** [review]

**Review by sergei-solonitcyn — APPROVED** [review]

**Comment by jrhizor** [comment]
Good idea with the prefix. Switched to that instead. Once this passes I'll merge.

File: airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java [hunk 2]
```diff
@@ -121,15 +124,17 @@
...
   private final LogConfigs logConfigs;
   private final CloudStorageConfigs stateStorageCloudConfigs;
 
   public EnvConfigs() {
-    this(System::getenv);
+    this(System::getenv, () -> System.getenv().keySet());
   }
 
-  public EnvConfigs(final Function<String, String> getEnv) {
+  public EnvConfigs(final Function<String, String> getEnv, final Supplier<Set<String>> getAllEnvKeys) {
```
Comment (tuliren) [inline airbyte-config/models/src/main/java/io/airbyte/config/EnvConfigs.java:135]: Nitpick. Can you add Javadoc to this class to explain the prefix setup?

**Comment by jrhizor** [comment]
Good call on the Javadoc. After realizing I didn't have a good way to explain it I cleaned up the interface instead.

**Pull request merged**